Automattic/wp-calypso
# Add hooks and HOCs to `lib/viewport`

#### Changes proposed in this Pull Request

* Fix a bug in listener removal in `lib/viewport`
* Add tests for `lib/viewport`
* Add helper hooks and higher-order components to `lib/viewport/react-helpers`.
* Add tests for `lib/viewport/react-helpers` hooks and HOCs.

#### Testing instructions

* Ensure that the modified components making use of the helpers still work. It may be difficult to trigger the reactive behaviour for some of them.
* Ensure the new tests pass.

CC @jsnajdr: I think I found a reasonable way of testing hook (and HOC) reactivity. What do you think?
